### PR TITLE
configure sql connection timeouts

### PR DESCRIPTION
**Pull request by garbageslam**
this may help when the server loses connection to the database

---

```
    make sql timeouts configurable in a standard way
    
    this makes the sql config params a struct_opt object,
    which all servers can incorporate into their config objects
    
    this also pays down some tech debt and code duplicaiton where
    lots of "struct opt parsing" type functions are copy pasted
    everywhere, instead they are put in a crate called `mc-util-parse`.
```

### CODE REVIEW

File: fog/sql_recovery_db/src/lib.rs [hunk 1]
```diff
@@ -53,6 +57,34 @@ pub const SQL_MAX_PARAMS: usize = 65000;
 /// Maximal number of rows to insert in one batch.
 pub const SQL_MAX_ROWS: usize = 5000;
 
+/// SQL recovery DB connection configuration parameters
+#[derive(Debug, Clone, StructOpt, Serialize)]
+pub struct SqlRecoveryDbParams {
+    /// idle timeouts
+    #[structopt(long, env, default_value = "60", parse(try_from_str=parse_duration_in_seconds))]
+    postgres_idle_timeout: Duration,
+    /// max lifetime
```
Comment (eranrund) [inline fog/sql_recovery_db/src/lib.rs:66]: Nit: Can you please put new lines between fields, and capitalize the first letter? I believe that has been the convention in other places.

File: util/parse/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+/// as other stuff.
+pub fn load_css_file(filename: &str) -> Result<CssSignature, String> {
+    let bytes =
+        fs::read(filename).map_err(|err| format!("Failed reading file '{}': {}", filename, err))?;
+    let signature = CssSignature::try_from(&bytes[..])
+        .map_err(|err| format!("Failed parsing CSS file '{}': {}", filename, err))?;
+    Ok(signature)
+}
+
+/// Helper to format a sequence as a comma-separated list
```
Comment (eranrund) [inline util/parse/src/lib.rs:29]: I don't feel strongly about this, but this doesn't seem to belong in the parsing module... if it's only used by Ingest, why did it move here?

**Comment by eranrund** [comment]
This is nice. I like having a crate for common parsing stuff.

**Review by jgreat — APPROVED** [review]
LGTM, I see the options.  

Do we maintain documentation for the config option someplace?

File: util/parse/src/lib.rs (hunk 2 above)
Comment (garbageslam) [reply util/parse/src/lib.rs:29]: i think i might use it other places, idk, i think this is the most idiomatic way right now to display a comma separated list of items in a container. it was weird to have it be like a formatting utility private to ingest server

i could move it back i guess

**Comment by garbageslam** [comment]
@jgreat i tried putting much better documentation and links in the `--help` this time

**Review by eranrund — APPROVED** [review]

**Pull request merged**